### PR TITLE
[CG-8871] fix: removing dead code removes code that should stay 

### PR DESCRIPTION
**Pull request by tkfoss**
# Motivation

# Content

# Testing

# Please check the following before marking your PR as ready for review

- [ ] I have added tests for my changes
- [ ] I have updated the documentation or added new documentation as needed

### CODE REVIEW

**Review by bagel897 — CHANGES_REQUESTED** [review]
Can we not add a new edge type? That adds a lot of complexity and will likely impact performance. Instead, can we fix the existing edges?

File: src/codegen/sdk/python/assignment.py [hunk 1]
```diff
@@ -96,3 +99,70 @@ def inline_comment(self) -> PyCommentGroup | None:
...
+
+        Args:
+            delete_formatting (bool): Whether to delete surrounding whitespace and formatting. Defaults to True.
+            priority (int): Priority of the removal transaction. Higher priority transactions are executed first. Defaults to 0.
+            dedupe (bool): Whether to deduplicate removal transactions at the same location. Defaults to True.
+
+        Returns:
+            None
+        """
+        if getattr(self.parent, "assignments", None) and len(self.parent.assignments) > 1:
```
Comment (bagel897) [inline src/codegen/sdk/python/assignment.py:119]: Can we do isinstance here?

File: src/codegen/sdk/core/import_resolution.py [hunk 2]
```diff
@@ -607,6 +607,8 @@ def names(self) -> Generator[tuple[str, Self | WildcardImport[Self]], None, None
                     imp.file.invalidate()
 
                 return
+            elif self.resolved_symbol is None:
```
Comment (bagel897) [inline src/codegen/sdk/core/import_resolution.py]: Can we break this part out into it's own pr with a test case. This specific change looks good

File: src/codegen/sdk/python/assignment.py [hunk 3]
```diff
@@ -96,3 +99,70 @@ def inline_comment(self) -> PyCommentGroup | None:
         """
         # HACK: This is a temporary solution until comments are fixed
         return PyCommentGroup.from_symbol_inline_comments(self, self.ts_node.parent)
+
+    @remover
+    def remove(self, delete_formatting: bool = True, priority: int = 0, dedupe: bool = True) -> None:
```
Comment (bagel897) [inline src/codegen/sdk/python/assignment.py:104]: Maybe it'll be better to implement _smart_remove rather than overwriting remove?

File: src/codegen/sdk/python/import_resolution.py [hunk 4]
```diff
@@ -232,6 +244,26 @@ def from_future_import_statement(cls, import_statement: TSNode, file_node_id: No
             imports.append(imp)
         return imports
 
+    @reader
+    @noapidoc
+    @override
+    def _resolved_types(self) -> Generator[ResolutionStack[Self], None, None]:
```
Comment (bagel897) [inline src/codegen/sdk/python/import_resolution.py:251]: I don't think this is the right approach. It will create duplicated edges. For example
```python
def foo():
     ...
def bar():
     ...
```
```python
from filea import *
foo() # will be a usage of foo and bar
```


For some context, we regular imports are resolved as follows.
1. We add the import to valid_import_names
2. When resolving, we check valid_symbol_names for the name
3. We call _resolved_types on the import

Since it's impossible to know which symbol you're importing in a wildcard, wildcards follow the following pattern
1. We check the imported file and get all valid_import_names (this is the same as valid_symbol_names in python) from that file
2. For each imported symbol, we created a wilcardImport for that name and put it in the valid imports
3. When resolving, we call _resolved_types on the wildcard import. We create an edge through the import but we point it at the specific imported symbol. 

For example, in the above case, we'd call _resolved_types on the wildcard for foo. But it'll create edges from the import to foo and from the call site to foo

**Review by bagel897 — CHANGES_REQUESTED** [review]
We don't have the full testing required to make some of these changes with confidence. Can we add a lot of tests for edge cases and maybe split this up?

File: src/codegen/sdk/python/assignment.py (hunk 3 above)
Comment (tkfoss) [reply src/codegen/sdk/python/assignment.py:104]: I need to know what Assignment it's being called on and remove dissolves it into Value which does not preserve the distinction, if there is a better idea than override let me know

File: src/codegen/sdk/python/import_resolution.py (hunk 4 above)
Comment (tkfoss) [reply src/codegen/sdk/python/import_resolution.py:251]: It seemed to work as expected, I ve added additional UTs, have I missed anything?

**Comment by tkfoss** [comment]
addressed in https://github.com/codegen-sh/codegen-sdk/pull/611 and https://github.com/codegen-sh/codegen-sdk/pull/612
